packages/ipfs-message-port-server/README.md

```diff
@@ -0,0 +1,103 @@
+[![](https://img.shields.io/badge/freenode-%23ipfs-blue.svg?style=flat-square)](http://webchat.freenode.net/?channels=%23ipfs)
+[![Travis CI](https://flat.badgen.net/travis/ipfs/js-ipfs)](https://travis-ci.com/ipfs/js-ipfs)
+[![Codecov branch](https://img.shields.io/codecov/c/github/ipfs/js-ipfs/master.svg?style=flat-square)](https://codecov.io/gh/ipfs/js-ipfs)
+[![Dependency Status](https://david-dm.org/ipfs/js-ipfs/status.svg?path=packages/ipfs-message-port-server)](https://david-dm.org/ipfs/js-ipfs?path=packages/ipfs-message-port-server)
+[![js-standard-style](https://img.shields.io/badge/code%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/feross/standard)
+
+> A library for providing IPFS node over [message channel][]. This library enables
+applications running in the different JS context to use [IPFS API](https://github.com/ipfs/js-ipfs/tree/master/docs/api) (subset) via `ipfs-message-port-client`.
+
+
+## Lead Maintainer <!-- omit in toc -->
+
+[Alex Potsides](https://github.com/achingbrain)
+
+## Table of Contentens <!-- omit in toc -->
+
+- [Install](#install)
+- [Contribute](#contribute)
+- [License](#license)
+
+## Install
+
+```bash
+$ npm install --save ipfs-message-port-server
+```
+
+## Usage
+
+This library can wrap JS IPFS node and expose it over the [message channel][].
+It assumes `ipfs-message-port-client` on the other end, however it is not
+strictly necessary anything compling with wire protocol will do.
+
+It provides following API subseset:
+
+- [`ipfs.dag`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/DAG.md)
+- [`ipfs.block`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/BLOCK.md)
+- [`ipfs.add`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/FILES.md#ipfsadddata-options)
+- [`ipfs.cat`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/FILES.md#ipfscatipfspath-options)
+- [`ipfs.files.stat`](https://github.com/ipfs/js-ipfs/blob/master/docs/core-api/FILES.md#ipfsfilesstatpath-options)
+
+Server is designed to run in the [SharedWorker][] (although it is possible to
+run it in the other JS contexts). Example below illustrates running js-ipfs
+node in [SharedWorkr][] and exposing it to all connected ports
+
+```js
+const IPFS = require('ipfs')
+const { IPFSService } = require('ipfs-message-port-server')
+const { Server } = require('ipfs-message-port-server/src/server')
+
+const main = async () => {
+  const ports = []
+  // queue connections that occur while node was starting.
+  self.onconnect = ({ports}) => connections.push(...ports)
+
+  const ipfs = await IPFS.create()
+  const service = new IPFSService(ipfs)
+  const server = new Server(service)
+
+  // connect new ports and queued ports with the server.
+  self.onconnect = ({ports}) => server.connect(ports[0])
+  for (const port of ports.splice(0)) {
+    server.connect(port)
+  }
+}
+
+main()
+```
+
+
+### Additional Consideration
```

Comment:
This section should be in TOC, for easy discovery when skimming (similar note in `README` for `-client`)
```suggestion
### Performance Consideration
```